tiny_dnn/core/framework/tensor.h

```diff
@@ -335,7 +318,36 @@ class Tensor {
                                                           shape_.end()));
     }
 
+    Tensor view(std::initializer_list<size_t> const &start,
+                std::initializer_list<size_t> const &shape) {
+	if (start.size() > kDimensions || shape.size() > kDimensions) {
+            throw nn_error("Overpassed number of existing dimensions.");
+	}
+        return Tensor(storage_ptr_, compute_offset(start, shape_), shape);
+    }
+
+    bool isView() const {
+        return size_ != storage_ptr_->size();
```

Comment:
Oh. Actually my comment didn't come through (was still "pending"). Sorry, I'm catching up with this "modern" tools ;)
size for vectors is just a pointer difference, probably inlined, I wouldn't bother about speed here.